core/trino-main/src/main/java/io/trino/sql/analyzer/Analyzer.java

```diff
@@ -92,15 +100,30 @@ public Analysis analyze(Statement statement, QueryType queryType)
         analyzer.analyze(rewrittenStatement, Optional.empty());
 
         // check column access permissions for each table
-        analysis.getTableColumnReferences().forEach((accessControlInfo, tableColumnReferences) ->
+        analysis.getTableColumnReferencesInStatements().forEach((accessControlInfo, tableColumnReferences) ->
                 tableColumnReferences.forEach((tableName, columns) ->
-                        accessControlInfo.getAccessControl().checkCanSelectFromColumns(
-                                accessControlInfo.getSecurityContext(session.getRequiredTransactionId(), session.getQueryId()),
-                                tableName,
-                                columns)));
+                        checkColumnsAccess(accessControlInfo, tableName, columns)));
         return analysis;
     }
 
+    private void checkColumnsAccess(Analysis.AccessControlInfo accessControlInfo, QualifiedObjectName tableName, Set<String> columns)
+    {
+        SecurityContext securityContext = accessControlInfo.getSecurityContext(session.getRequiredTransactionId(), session.getQueryId());
+
+        if (SystemSessionProperties.isHideInaccesibleColumns(session)) {
```

Comment:
I think filtering should happen during the semantic analysis phase where relations are being analyzed. Here we should simply check the access as we used to do it.

Reply:
@kokosing 
What do you think if I'll move the implementation in the `io.trino.security.AccessControl` interface as a default method?
The method simply change the type of exception in case of direct access to an hidden column from "Access Denied" to "Column Not Found" if configured.

```
 default void checkColumnsAccess(SecurityContext context, QualifiedObjectName tableName, Set<String> columns, boolean hideInaccessibleColumns)
    {
        if (hideInaccessibleColumns) {
            Set<String> accessibleColumns = filterColumns(context, tableName.asCatalogSchemaTableName(), columns);
            Set<String> notAccessibleColumns = columns.stream().filter(input -> !accessibleColumns.contains(input)).collect(Collectors.toSet());
            if (!notAccessibleColumns.isEmpty()) {
                throw new TrinoException(COLUMN_NOT_FOUND, String.format("Columns %s cannot be resolved", notAccessibleColumns));
            }
        }
        checkCanSelectFromColumns(
                context,
                tableName,
                columns);
    }
```

Reply:
No. It should be part of semantic analysis.

Reply:
Can you suggest me where to add the check? I tried to add it in the ExpressionAnalyzer (i.e. in visitIdentifier method) but, in that case, I cannot use the hidden column in the row filter and, as described I need at the same time to hide the column but apply the row filter on it.

Reply:
I was thinking about `io.trino.sql.analyzer.StatementAnalyzer.Visitor#analyzeSelectAllColumns`

Reply:
The filter is already applied there in the PR. The problem is not where the selectAll is resolved but when I try to access to the single column: It must be allowed in the row filter but forbidden otherwhere.

Reply:
The filter is a kind of a view. As you see there is a `io.trino.spi.security.ViewExpression` to represent this. This has an user who is an "owner" of the "view" so filter should resolved using that user and user should be able to access columns used in row filters.

I am under impression that you are solving two things. (1) Filtering restricted columns when `SELECT *` is used, which I like. (2) skip authorization checks for row filters, which is tricky and I basically I would prefer to avoid it. 
How about solving one issue at a time and let's first tackle (1)?

Reply:
@findepi commented that it is currently possibile https://github.com/trinodb/trino/pull/9991#issuecomment-979784880, is it? Moreover in my use case that I describe here: https://github.com/trinodb/trino/pull/9991#issuecomment-979451533t the 2 things make sense only together. 

IMO It is not a trick for skipping authentication checks, it is a way for defining checks on rows and columns in independent way so that I can choose for a user which columns and row she can see.

Reply:
@kokosing In any case I can remove this piece of code: the only side effect is that you'll received an "Access Denied" instead a "column not resolved" exception as it is now. Could be acceptable for you?